pkg/cri/server/sandbox_run.go

```diff
@@ -139,33 +139,7 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 			return nil, fmt.Errorf("failed to create network namespace for sandbox %q: %w", id, err)
 		}
 		sandbox.NetNSPath = sandbox.NetNS.GetPath()
-		defer func() {
-			if retErr != nil {
-				deferCtx, deferCancel := ctrdutil.DeferContext()
-				defer deferCancel()
-				// Teardown network if an error is returned.
-				if err := c.teardownPodNetwork(deferCtx, sandbox); err != nil {
-					log.G(ctx).WithError(err).Errorf("Failed to destroy network for sandbox %q", id)
-				}
 
-				if err := sandbox.NetNS.Remove(); err != nil {
-					log.G(ctx).WithError(err).Errorf("Failed to remove network namespace %s for sandbox %q", sandbox.NetNSPath, id)
-				}
-				sandbox.NetNSPath = ""
-			}
-		}()
-
-		// Setup network for sandbox.
-		// Certain VM based solutions like clear containers (Issue containerd/cri-containerd#524)
-		// rely on the assumption that CRI shim will not be querying the network namespace to check the
-		// network states such as IP.
-		// In future runtime implementation should avoid relying on CRI shim implementation details.
-		// In this case however caching the IP will add a subtle performance enhancement by avoiding
-		// calls to network namespace of the pod to query the IP of the veth interface on every
-		// SandboxStatus request.
-		if err := c.setupPodNetwork(ctx, &sandbox); err != nil {
-			return nil, fmt.Errorf("failed to setup network for sandbox %q: %w", id, err)
-		}
 		sandboxCreateNetworkTimer.UpdateSince(netStart)
```

Comment:
We update the metrics twice now, is that expected?

Reply:
Yes, after this change, the code of CreateNetwork are in two places. So it is correct to sum them up. That is the same logic as before.

Reply:
we've split up the ns part and the setup .. perhaps we should have two network timers..

Reply:
@mikebrow Not sure that is the desired metrics granularity. What do you think?

Reply:
the current granularity is ns+setup.. I'd either split it into a new metric for ns(s) for the pod or just do it after setup like it is now..

Reply:
This is going to get more complicated soon..  let's just take out the portion separated for the ns creation.. and document somewhere that this is CNI setup timing  (not namespaces for the pod)

See expected user ns pattern:
```
- configure all needed namespaces for the sandbox/infra container, including the user namespace
- call runtime to start the infra container and creates all needed namespaces
- call CNI to initialize pod network with the net ns created by the runtime for the infra container
```

Reply:
Yes, #6909 works in the described above, to solve an issue related to user namespace.

Reply:
ok, I will separate the metrics then.